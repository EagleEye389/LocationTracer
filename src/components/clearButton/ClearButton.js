import React from "react";
import PropTypes from "prop-types";

import "./clearButton.css";

/**
 * @description Cross button renders X if the props.value is not null
 */
const ClearButton = props => {
  const { value, onChangeInput, name } = props;
  return (
    <div
      className={`input-group-addon ${value && "visible"} `}
      onClick={() => onChangeInput("", name)}
    >
      X
    </div>
  );
};

ClearButton.defaultProps = {
  value: "default"
};

ClearButton.propTypes = {
  value: PropTypes.bool,
  name: PropTypes.string.isRequired,
  onChangeInput: PropTypes.func.isRequired
};

export default ClearButton;
