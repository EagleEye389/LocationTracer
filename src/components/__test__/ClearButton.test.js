import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import ClearButton from "../clearButton/ClearButton";

let component;
describe("Clearbutton is loaded properly", () => {
  beforeEach(() => {
    component = shallow(
      <ClearButton value={true} name="destination" onChangeInput={jest.fn()} />
    );
  });

  it("Should visible", () => {
    expect(component.find("div").hasClass("visible")).toBeTruthy();
  });

  it("Should not visible", () => {
    component.setProps({ value: false });
    expect(component.find("div").hasClass("visible")).toBeFalsy();
  });
});

describe("Clearbutton snapshot", () => {
  it("Should render correctly", () => {
    const tree = renderer
      .create(
        <ClearButton
          value={true}
          name="destination"
          onChangeInput={jest.fn()}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Should not render", () => {
    const tree = renderer
      .create(
        <ClearButton
          value={false}
          name="destination"
          onChangeInput={jest.fn()}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
