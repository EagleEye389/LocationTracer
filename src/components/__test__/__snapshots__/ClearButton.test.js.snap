// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clearbutton snapshot Should not render 1`] = `
<div
  className="input-group-addon false "
  onClick={[Function]}
>
  X
</div>
`;

exports[`Clearbutton snapshot Should render correctly 1`] = `
<div
  className="input-group-addon visible "
  onClick={[Function]}
>
  X
</div>
`;
